### PR TITLE
Display Custom Blocks Listing in the Second Tab of the Left Sidebar

### PR DESCRIPTION
**Pull request by omar-bear**
#### Description:
Integrate the listing of Personalized Blocks in the existing second tab in the left sidebar when the "Blocks" button in the page header is selected.

#### Frontend:

1. **Initialize Main Vue.js Component `PersonalizedBlocksPlugin`**
   - File: `personalizedBlocksPlugin.js`
   - Task:
     - Create a Vue.js component to configure the ViewModel and extensions. Use `this.vm.` to refer to the ViewModel.
   - Example Template:
     ```javascript
     Vue.component('PersonalizedBlocksPlugin', {
       components: {
         PersonalizedBlocksListComponent,
       },
       data: () => ({
         viewModel: vm,
       }),
       //... other logic
     });
     ```

2. **Initialize List Component `PersonalizedBlocksListComponent`**
   - File: `personalizedBlocksListComponent.js`
   - Task:
     - Create a Vue.js component to manage the listing of Personalized Blocks, including API calls, loading state, and error handling.
   - Example Template:
     ```javascript
     Vue.component('PersonalizedBlocksListComponent', {
       //... other logic
     });
     ```

3. **API Calls to Fetch Personalized Blocks**
   - File: `personalizedBlocksListComponent.js`
   - Task:
     - Add a method named `fetchPersonalizedBlocks` that will make an API call to the `getPersonalizedBlocks` endpoint. Utilize `this.vm.metadata.groupId` for the `groupId`.

4. **Manage Loading State**
   - File: `personalizedBlocksListComponent.js`
   - Task:
     - Add a Vue.js data property `isLoading` to manage the loading state. Display a loading spinner while fetching data.

5. **Vue Data Property for Personalized Blocks**
   - File: `personalizedBlocksListComponent.js`
   - Task:
     - Add a Vue.js data property named `personalizedBlocks` to store the fetched blocks.

6. **Displaying Personalized Blocks Using Existing HTML Structure**
   - File: `toolbox.tmpl.html`
   - Task:
     - Utilize the HTML structure already in place for displaying template blocks. Bind this part to the `personalizedBlocks` data property from the Vue component.

7. **Handle Errors**
   - File: `personalizedBlocksListComponent.js`
   - Task:
     - In case of an API failure, display an error toast notification using `this.vm.notifier.error(message)`.

8. **Integrate the Vue.js Component into HTML**
   - File: `toolbox.tmpl.html`
   - Task:
     - Add a tag `<personalized-blocks-list-component />` to integrate the Vue.js component within the HTML file.

#### Acceptance Criteria:

- The second tab in the left sidebar successfully displays the list of Personalized Blocks when `blocksActiveTab` is set to `CUSTOM_BLOCKS`.
- The HTML structure for displaying Personalized Blocks is the same as that for displaying template blocks.
- A loading spinner is displayed while fetching data.
- Toast notifications are displayed in case of API errors.

#### Time Estimate:
2-3 days

## RESULT

https://github.com/Badsender-com/LePatron.email/assets/80390318/a22a8e47-ca23-4155-98ab-771b36dd6b8a

### CODE REVIEW

File: public/lang/badsender-fr.js [hunk 1]
```diff
@@ -110,4 +110,10 @@ module.exports = {
   'saving-block': 'Enregistrement en cours',
   'save-block-success': 'Bloc enregistré avec succès',
   'save-block-error': "Erreur lors de l'enregistrement du bloc",
+
+  // PersonalizedBlocksListComponent translations
+  'personalized-blocks-fetch-error':
+    "Une erreur s'est produite lors de la récupération des Blocs perso.",
```
Comment (yoannfleurydev) [inline public/lang/badsender-fr.js:116]:
```suggestion
    "Une erreur s'est produite lors de la récupération des blocs perso.",
```

File: public/lang/badsender-en.js [hunk 2]
```diff
@@ -109,4 +109,10 @@ module.exports = {
   'saving-block': 'Saving block',
   'save-block-success': 'Block saved successfully',
   'save-block-error': 'Error saving block',
+
+  // PersonalizedBlocksListComponent translations
+  'personalized-blocks-fetch-error':
+    'An error occurred while fetching Custom blocks.',
```
Comment (yoannfleurydev) [inline public/lang/badsender-en.js:115]:
```suggestion
    'An error occurred while fetching custom blocks.',
```

File: packages/editor/src/css/style_mosaico_tools.less [hunk 3]
```diff
@@ -798,11 +798,22 @@
...
-  .coming-soon {
-    font-style: italic;
+  /* CSS for the PersonalizedBlocksListComponent */
+
+  /* Style for the loading state */
+  #personalized-blocks-list .loading-state {
+    text-align: center;
+    font-size: 1.2em;
+    color: #888;
+    padding: 20px;
```
[the remark this answers is not in the document]
Comment (omar-bear) [reply packages/editor/src/css/style_mosaico_tools.less:808]: yes I think I will fix that !

**Review by DecampsRenan — APPROVED** [review]

File: packages/editor/src/js/vue/components/personalized-blocks-list/personalized-blocks-list-component.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        const groupId = this.vm?.metadata?.groupId;
+
+        axios
+          .get(getPersonalizedBlocks(), { params: { groupId } })
+          .then((response) => {
+            this.vm.personalizedBlocks(
+              response.data?.items.map(({ content, ...blockInformation }) => ({
+                ...content,
+                blockInformation,
+                id: "" // this is so important because it will be override when the block is added to the mail content
```
[the remark this answers is not in the document]
Comment (omar-bear) [reply packages/editor/src/js/vue/components/personalized-blocks-list/personalized-blocks-list-component.js:41]: I mean adding `id: ""` is important, i will try to clarify my comment

**Pull request merged**